pandas/core/groupby/generic.py

```diff
@@ -609,12 +609,35 @@ def value_counts(
 
         names = self.grouper.names + [self.obj.name]
 
-        if is_categorical_dtype(val.dtype) or (
-            bins is not None and not np.iterable(bins)
-        ):
+        if is_categorical_dtype(val.dtype):
```

Comment:
can we just call the appropriate implementation. creating a new grouper and calling like this just obfuscatest he code. would really prefer simply to have a single implementation and just call it.

Reply:
I understand why it it obfuscates the code, and I wonder what would be the approriate implementation here. Do you mean introducing a method `_groupby_value_counts` containing most of the current DataFrame implementation logic, that would be called by both Series and DataFrame implementation ?

Reply:
@jreback could you please elaborate what you think would be the approriate implementation to call here ?

Reply:
need to refactor the value counts to a separate function then just call it for frame and series